### ISSUE TITLE
How about adding a note of `_get_optimizer` in "Customize what happens in Model.fit" guide?

### ISSUE THREAD

**Issue by jeongukjae** [tensorflow/tensorflow#49368]
## URL(s) with the issue:

https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit#wrapping_up_an_end-to-end_gan_example

### Clear description

When using Keras mixed-precision API and overriding `tf.keras.Model.compile` function like the above guide, training loss cannot converge easily, because the optimizers are not wrapped in `LossScaleOptimizer`. I found it can be fixed using `tf.keras.Model._get_optimizer` function like below.

https://github.com/tensorflow/tensorflow/blob/v2.5.0/tensorflow/python/keras/engine/training.py#L479-L586

I think it would be much better to add a note like ``it is recommended to wrap optimizer objects using `self._get_optimizer` when using Keras mixed-precision API`` in that guide.

**Comment by Saduf2019** [comment]
A pr has been created once the [pr](https://github.com/keras-team/keras-io/pull/524) is merged this issue will be moved to closed status.

**Comment by Saduf2019** [comment]
@jeongukjae 
Moving this issue to closed status as per update on pr.
